### ISSUE TITLE
Thoughts on syntax parity with graphql-import?

### ISSUE THREAD

**Issue by lennyburdette** [csantero/broccoli-graphql-filter#3]
The graphql-import library uses a similar but incompatible syntax.

#### broccoli-graphql-filter
```graphql
#import from 'some/runtime/path'
```

#### [graphql-import](https://oss.prisma.io/content/graphql-import/overview)
```graphql
# import FragmentName from 'some/node-resolvable/path'
```

I'm interested in parity because I would like to use tools like `apollo client:extract` on the queries in my Ember app. Currently they fail with a syntax error.

This is obviously not a trivial change. I can foresee two issues:
1. The path in the broccoli `#import` statement doesn't map to a file on disk so graphql-import couldn't use it anyway. (graphql-import uses [resolve-from](https://github.com/sindresorhus/resolve-from) to find files.) We'd need a way to teach graphql-import about Ember's file system layout, wait for [ef4's new builder spec](https://github.com/embroider-build/embroider) to land, or ...?.
2. The graphql-import syntax matches ES6 module imports, including import specifiers. This is more complicated than broccoli-graphql-filter, which is effectively the same as `import * as AllFragments from 'path'`. Dealing with aliases (`import { foo as bar }`) is considerably more complicated. 

   This project could limit itself to supporting only `* as AllFragments` for simplicity's sake.

Thoughts? I wish I had a real solution for point number 1!

**Comment by csantero** [comment]
@lennyburdette I'm personally fine with switching to the new syntax, in the interest of broader GraphQL ecosystem compatibility. But since the main consumer of broccoli-graphql-filter is ember-apollo-client, I'm going to defer to @bgentry.

**Comment by bgentry** [comment]
I'm fine with it, though I don't have much time to contribute to the effort atm. We're close to migrating ember-apollo-client to TypeScript which may already carry with it a major version bump, so that would be a good time to make this breaking change.

**Comment by dfreeman** [comment]
Independent of point 1 (which is tricky, but you're right that the Embroider stuff is likely to help), I think switching to named imports is a good idea.

I've been playing a bit with type declaration generation for GraphQL queries, and was planning to propose this switch in support of that for largely the same reasons—it gives the missing information that automated tooling needs to be able to piece everything together.

Edit: looking more closely at `graphql-import`, I don't love their syntax: it's pretty explicitly _not_ the same as ES import syntax, which is unfortunate (technically our `import 'some-path'` is a valid ES import today). It seems like they've got a stake in the ground already, though, so it may not make sense for us to push ES-style `import { MyFragment, Other as LocalName } from '...'` imports here.

**Comment by dfreeman** [comment]
@lennyburdette does `apollo client:extract` actually follow `# import` directives? I can't find any reference to that in their documentation.

**Comment by lennyburdette** [comment]
I think you're right. I'm getting the graphql-cli (by prisma, who also makes graphql-import) confused with apollo-cli. 

I guess there's no hurry here since the tooling ecosystem is so fragmented. Thanks for the quick replies though!

**Comment by csantero** [comment]
This was implemented in #9.
